### PR TITLE
Adsk contrib - Make the interchange roles required for future config versions

### PR DESCRIPTION
**Pull request by cedrik-fuoco-adsk**
Hi,

This PR covers the groundwork for issue #1639. For now, the validation will not reject a version 2.2 config that doesn't have the interchange roles and the other mandatory roles. Instead, it will log a new error message (still warning level) in order for the validation to not fail.

Ociocheck has been refactored a bit and updated with additional checks. See Doug's [comment ](https://opencolorio.slack.com/archives/C3NLV668M/p1665725355036919) on Slack.

Since this feature modifies ociocheck quite a lot, the isArchivable part (main.cpp, lines 498, 506 and 533) will be added back once OCIOZ Archive config is merged on the main branch. This is because both features modify ociocheck, but the modifications here will include both. Therefore, an extra commit will be needed in order to do that here.

### CODE REVIEW

File: src/apps/ociocheck/main.cpp [hunk 1]
```diff
@@ -161,182 +214,324 @@ int main(int argc, const char **argv)
...
-                                        OCIO::ROLE_COLOR_PICKING,
-                                        OCIO::ROLE_TEXTURE_PAINT, OCIO::ROLE_MATTE_PAINT,
-                                        OCIO::ROLE_RENDERING,
-                                        OCIO::ROLE_INTERCHANGE_SCENE,
-                                        OCIO::ROLE_INTERCHANGE_DISPLAY,
-                                        NULL };
-            int MAXROLES=256;
-            for(int i=0;i<MAXROLES; ++i)
+            // All roles defined in OpenColorTypes.h.
+            std::set<std::string> allRolesSet = {
```
[the remark this answers is not in the document]
Comment (cedrik-fuoco-adsk) [reply src/apps/ociocheck/main.cpp:218]: I agree. I will change it.

File: tests/cpu/Config_tests.cpp [hunk 2]
```diff
@@ -314,6 +314,270 @@ OCIO_ADD_TEST(Config, roles)
...
+    {
+        // Test that aces_interchange role has the wrong colorspace type.
+
+        // Set a display-referred colorspace to both interchange roles.
+        config->setRole(OCIO::ROLE_INTERCHANGE_SCENE, dcs->getName());
+        config->setRole(OCIO::ROLE_INTERCHANGE_DISPLAY, dcs->getName());
+
+        OCIO::LogGuard logGuard;
+        OCIO_CHECK_NO_THROW(config->validate());
+        std::cout << "Log => [" << logGuard.output() << "]" << std::endl;
```
[the remark this answers is not in the document]
Comment (cedrik-fuoco-adsk) [reply tests/cpu/Config_tests.cpp:536]: Good catch. Indeed, it is a debug print. I'll remove it.

File: tests/cpu/Config_tests.cpp [hunk 3]
```diff
@@ -314,6 +314,270 @@ OCIO_ADD_TEST(Config, roles)
...
+    config->setRole(OCIO::ROLE_INTERCHANGE_SCENE, scs->getName());
+    config->setRole(OCIO::ROLE_INTERCHANGE_DISPLAY, dcs->getName());
+
+    {
+        // Check that no warning is logged when all required roles are set.
+
+        OCIO::LogGuard logGuard;
+        OCIO_CHECK_NO_THROW(config->validate());
+        OCIO_CHECK_ASSERT(
+        StringUtils::StartsWith(logGuard.output(), ""));
```
[the remark this answers is not in the document]
Comment (cedrik-fuoco-adsk) [reply tests/cpu/Config_tests.cpp:420]: It should not have any other logs, but I think that testing if the log doesn't contain [OpenColorIO Error], is a good idea.
I think it makes more sense in that context.

**Review by remia — APPROVED** [review]

**Comment by cedrik-fuoco-adsk** [comment]
I've updated ociocheck to include the archive-related code since that feature has been merged.

Please look at the second last commit https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1702/commits/739b8723225cfe58958b6a6bb2d21c8a1d8b93ee

The latest is the rebase from main.

**Comment by cedrik-fuoco-adsk** [comment]
A new PR has been created to clean up the PR from the rebase commits and to fix the issue with the DCO.

Please look here : https://github.com/AcademySoftwareFoundation/OpenColorIO/pull/1711

@remia @michdolan

**Comment by doug-walker** [comment]
Closing this PR, please refer to PR #1711 instead.

**Pull request closed without merging**